asaps/models.py

```diff
@@ -108,24 +108,34 @@ def filter_note_type(client, csv_data, rec_obj, note_type, operation,
         try:
```

Comment:
What is the try/except catching? Do some notes not have a `type`? Or do some notes not have a `subnotes`? Or both? Or are there other keys this is covering?

Reply:
Both are evidently optional as I recall, I had to add this when I was testing and getting errors since they didn't exist in some notes

Reply:
In that case, there are a couple changes that can be made to remove a few levels of nesting and make the code a little easier to read and less likely to cause problems:

```python
for note in [n for n in rec_obj['notes'] if n.get('type') == note_type]:
    for subnote in note.get('subnotes'):
        ...
return rec_obj
```

It makes me a little uncomfortable having a try/except block for `KeyError` covering so many different key accesses on multiple dictionaries. There are a lot of opportunities here for real errors to get silently discarded.

Reply:
Let me actually take another look at the type field, I checked and it is required in the web interface so I may have just hit some bad records that were migrated incorrectly. This may be more of a data cleanup issue and then we could just worry about subnotes as the optional field